taichi/analysis/same_statements.cpp

```diff
@@ -191,34 +216,68 @@ class IRNodeComparator : public IRVisitor {
     }
   }
 
-  static bool run(IRNode *root1, IRNode *root2) {
-    IRNodeComparator comparator(root2);
+  static bool run(IRNode *root1,
+                  IRNode *root2,
+                  const std::optional<std::unordered_map<int, int>> &id_map,
+                  bool check_same_value) {
+    IRNodeComparator comparator(root2, id_map, check_same_value);
     root1->accept(&comparator);
     return comparator.same;
   }
 };
 
 namespace irpass::analysis {
-bool same_statements(IRNode *root1, IRNode *root2) {
+bool same_statements(
+    IRNode *root1,
+    IRNode *root2,
+    const std::optional<std::unordered_map<int, int>> &id_map) {
+  // When id_map is std::nullopt by default, this function tests if
```

Comment:
The doc is great! But i think it's better to put this to the headers, so that some IDEs can show it when we use `same_statements`.  Or at least could you add a comment in the header to explain the condition of when the check will be recursive? (IIUC, it depends on both  `id_map`, and whether roots are container? This can be confusing.)

Reply:
When roots are containers, it's natural that `same_statements` traverses the contents of the roots. Recursiveness only applies to statements outside the roots.

Reply:
Added some comments in `analysis.h`. Not sure if the doc format is correct though...